spec.bs

```diff
@@ -101,3 +264,79 @@ dictionary AuctionAdConfig {
   AbortSignal? signal;
 };
 </xmp>
+
+
+# Permissions Policy integration # {#permissions-policy-integration}
+
+This specification defines a [=policy-controlled feature=] identified by the string
+"<code><dfn noexport>join-ad-interest-group</dfn></code>". Its [=default allowlist=] is `'self'`.
+
+Note: In the Chromium implementation the [=default allowlist=] is temporarily set to `*` to ease
+testing.
+
+
+# Structures # {#structures}
+
+<h3 dfn-type=dfn>Interest group</h3>
+
+An interest group is a [=struct=] with the following items:
+
+<dl dfn-for="interest group">
+: <dfn>expiry</dfn>
+:: A point in time at which the browser will forget about this interest group.
+: <dfn>owner</dfn>
+:: An [=url/origin=].
+: <dfn>name</dfn>
+:: A [=string=].
+: <dfn>priority</dfn>
+:: A double. Defaulting to 0.0. Used to select which interest groups participate in an auction when
+  the number of interest groups are limited by {{AuctionAdConfig/perBuyerGroupLimits}}.
+: <dfn>enable bidding signals prioritization</dfn>
+:: A [=boolean=]. Being true if the interest group's priority should be calculated using vectors
+  from bidding signals fetch.
+: <dfn>priority vector</dfn>
+:: Null or an [=ordered map=] whose [=map/key|keys=] are [=strings=] and whose [=map/value|values=]
+  are doubles. Its dot product with the {{AuctionAdConfig/perBuyerPrioritySignals}} will be used in
+  place of [=interest group/priority=], if set.
+: <dfn>priority signals overrides</dfn>
+:: Null or an [=ordered map=] whose [=map/key|keys=] are [=strings=] and whose [=map/value|values=]
+  are doubles. Overrides the {{AuctionAdConfig}}'s corresponding priority signals.
+: <dfn>execution mode</dfn>
+:: An {{WorkletExecutionMode}}, defaulting to "compatibility".
+: <dfn>bidding url</dfn>
+:: Null or a [=URL=]. The URL to fetch the buyer's Javascript from.
+: <dfn>bidding wasm helper url</dfn>
+:: Null or a [=URL=]. Lets the bidder provide computationally-expensive subroutines in WebAssembly,
+  rather than JavaScript, to be driven from the JavaScript function provided by
+  [=interest group/bidding url=].
+: <dfn>daily update url</dfn>
+:: Null or a [=URL=]. Provides a mechanism for the group's owner to periodically update the
+  attributes of the interest group.
+: <dfn>trusted bidding signals url</dfn>
+:: Null or a [=URL=]. Provide a mechanism for making real-time data available for use at bidding
+  time.
+: <dfn>trusted bidding signals keys</dfn>
+:: Null or a [=list=] of [=string=].

```

Comment:
Is null different from an empty list? I noticed that you don't check whether `trustedBiddingSignalsKeys` is set above.

Reply:
That was my mistake. Added "If |group|["{{AuctionAdInterestGroup/trustedBiddingSignalsKeys}}"] [=list/exists=]" before setting this field.
So the nulls here in this struct are saying these fields are optional, such as _**array<string>? trusted_bidding_signals_keys**_ defined in our mojom for this field.